### DISCUSSION TITLE
Strange behavior with HR training programs

### DISCUSSION THREAD

**Discussion by grube386** [cagnulein/qdomyos-zwift#1182] [General]
Hi,

I have set up a test training program with basically just HR zones. The issue is that I cannot see the PID currently active in the PID tile and the Next rows tile shows N/A. 

Having gone through the whole program just now, the resistance behaves strangely. It is either almost full 14-15 or basically off 4-5. Even if I push it two zones over (the planned zone is 2 and I am at 4) it will take a minute or two to decrease the resistance.

The program is in the attachment.

Thank you so much for your attention. 
Matej
[testtraining - Copy.txt](https://github.com/cagnulein/qdomyos-zwift/files/10460117/testtraining.-.Copy.txt)

**Comment by cagnulein** [comment]
thanks @grube386 i will check tomorrow

**Reply by cagnulein** [reply]
@grube386 i fixed the issue about the fact you don't see the active PID heart zone in the tile and also the next rows thing.

**Reply by cagnulein** [reply]
the behaviour that you're describing it's about the looptimehr value. I mean it's too slow, because that value means that QZ will increase or decrease the resistance based on your zone every looptimehr seconds. Since your hr is changing not so quickly you will have strong increasing/decrising of the resistance.

If i can suggest, you should put the looptimehr to 30 seconds or similar, and when you want to step to a zone to another, put a resistance line directly on the train program so you can do directly a jump. than the pid zone will regulate it. What do you think?

If you want to try the fixed that I made in the tiles send me an email to roberto.viola83 at gmail.com indicating this ticket and if you are on android or iOS. Thanks

**Reply by grube386** [reply]
@cagnulein, glad to hear that you fixed the issue, you are fast 😄 I will send you an email for sure.
So if I define the resistance in the row, that will only be the initial resistance and then PID will take over? 

Also two questions or feature requests, I don't know if they already exist 😆 
1. A view on Android, that would include one graph and a few tiles would be amazing. 
2. Some sort of a sound indicating that a new row was started from the program for a mental reminder, would also be great. (At this point, I'll probably input a row dedicated for that, if the zone will go down, I will give a quick 30s row with low resistance and the opposite if the zone will go up)

**Reply by cagnulein** [reply]
> So if I define the resistance in the row, that will only be the initial resistance and then PID will take over?

Yes!

> A view on Android, that would include one graph and a few tiles would be amazing.

I already started a PR about this, but i need time to work on this :)

> Some sort of a sound indicating that a new row was started from the program for a mental reminder, would also be great. (At this point, I'll probably input a row dedicated for that, if the zone will go down, I will give a quick 30s row with low resistance and the opposite if the zone will go up)

please explain me more about this in a new ticket, i really like the idea of sound feedbacks

**Reply by grube386** [reply]
Do I open an issue or a new discussion here?

**Reply by cagnulein** [reply]
an issue as feature request is perfect, thanks

**Comment by grube386** [comment]
@cagnulein did my email perhaps get lost in your spam folder? Not trying to pressure you just wondering since you are always so fast with responses otherwise 😊

**Reply by cagnulein** [reply]
you did right because i don't see it, could you please send me again?
